lib/ampProtection.js

```diff
@@ -0,0 +1,70 @@
+/** @module tds */
+
+const {
+    generateDNRRule
+} = require('./utils')
+
+const AMP_PROTECTION_PRIORITY = 40000
+
+/**
+ * @typedef generateAmpProtectionRulesResult
+ * @property {import('./utils').DNRRule} rule
+ * @property {object} matchDetails
+ */
+
+/**
+ * Generator to produce the declarativeNetRequest rules and corresponding match
+ * details for the given trackerAllowlist configuration.
+ * @param {object} extensionConfiguration
+ *   The extension configuration.
+ * @param {function} isRegexSupported
+ *   A function compatible with chrome.declarativeNetRequest.isRegexSupported.
+ *   See https://developer.chrome.com/docs/extensions/reference/declarativeNetRequest/#method-isRegexSupported
+ * @return {AsyncGenerator<generateAmpProtectionRulesResult>}
+ */
+async function* generateAmpProtectionRules ({ features: { ampLinks } }, isRegexSupported) {
```

Comment:
I'm just using a generator function for consistency with the other rule generation libraries called by `extensionConfiguration.js`. I agree it doesn't add much, but I don't think it is especially worse either. I would rather leave this alone, but can change to return an array if you feel strongly?

Reply:
I noticed we use this pattern elsewhere, but that's not a reason to keep using it. I'd prefer an array return - it can still be used the same way in the config code.

Reply:
Fair enough, to my mind a generator makes sense here since we don't want the intermediary array. Waste of memory to generate it and then throw it away. But in practice, the number of rules generated is very short and it doesn't matter much. Done.